cmake/toolchain/gcc.x86.cmake

```diff
@@ -3,7 +3,7 @@
 ##  Copyright : EVE Project Contributors
 ##  SPDX-License-Identifier: BSL-1.0
 ##==================================================================================================
-set(CMAKE_C_COMPILER    gcc  )
-set(CMAKE_CXX_COMPILER  g++  )
+set(CMAKE_C_COMPILER    gcc-14  )
+set(CMAKE_CXX_COMPILER  g++-14  )
```

Comment:
Are we shipping many gccs?

Reply:
I think I only have 14 in my current CI image, and it is installed this way. I can later fix the image to use the default GCC selector indeed.